src/vellum/workflows/nodes/core/inline_subworkflow_node/tests/test_node.py

```diff
@@ -87,3 +89,30 @@ class OuterNode(InlineSubworkflowNode):
     # THEN we only have the outer node's outputs
     valid_events = [e for e in events if e.name == "bar"]
     assert len(valid_events) == len(events)
+
+
+def test_inline_subworkflow_node__base_inputs_validation():
+    """Test that InlineSubworkflowNode properly validates required inputs"""
+
+    # GIVEN a real subworkflow class with a required input
+    class SubworkflowInputs(BaseInputs):
+        required_input: str  # This is a required field without a default
+
+    class TestSubworkflow(BaseWorkflow[SubworkflowInputs, BaseState]):
+        pass
+
+    # AND a node that uses this subworkflow
+    class TestNode(InlineSubworkflowNode):
+        subworkflow = TestSubworkflow
+        subworkflow_inputs = {"required_input": None}
+
+    # WHEN we try to run the node
+    node = TestNode()
+
+    # THEN it should raise a WorkflowInitializationException
+    with pytest.raises(WorkflowInitializationException) as e:
+        list(node.run())
+
```

Comment:
[nit] so again, nodes ideally only raise `NodeException` and our various subworkflow nodes catch  `WorkflowInitializationException` in the implementation to raise the relevant `NodeException`